cloudinit/config/modules.py

```diff
@@ -265,16 +278,21 @@ def run_section(self, section_name):
         raw_mods = self._read_modules(section_name)
         mostly_mods = self._fixup_modules(raw_mods)
         distro_name = self.init.distro.name
+        schema = get_schema()
 
         skipped = []
         forced = []
         overridden = self.cfg.get("unverified_modules", [])
+        inapplicable_mods = []
         active_mods = []
-        for (mod, name, _freq, _args) in mostly_mods:
+        for module_details in mostly_mods:
+            (mod, name, _freq, _args) = module_details
             if mod is None:
                 continue
             worked_distros = mod.meta["distros"]
-
+            if not _is_applicable(module_details, self.cfg, schema=schema):
```

Comment:
I think we want to also consider that this new behavior may not be something we turn on by default on stable series Bionic/Focal/Impish/Jammy. I'd like us to think about a configuration option maybe under a `modules` top-level key that can be provided minimally in /etc/cloud/cloud.cfg(.d/*). Ideally, we might also want to surface that config-key in user-data too so folks could opt to `skip_modules_by_applicable_schema: true` or something during a specific instance launch. By default that config setting would be False on stable series because we don't want to change behavior.

Reply:
In a team standup conversation on this, the intent of this PR is that it is a performance improvement that consumers of stable releases will want turned on by default anyway.  We agree it should be writen in such a way as to not change behavior from the consumers point of view. As such, we'll ensure this is covered adequately by unit and integration tests to ensure no regressions are there. There is no need for us to provide a user-data or file-system top-level configuration key to turn this feature on or off because it should just work. If it does not, it it a bug that needs to be addressed in the impementation.